permission/permissions/permissions.go

```diff
@@ -0,0 +1,59 @@
+	CalendarAdmin               Permissions = "Calendar.Admin"
+	CalendarMeetingAdmin        Permissions = "Calendar.Meeting.Admin"
+	CalendarMeetingCreator      Permissions = "Calendar.Meeting.Creator"
+	CalendarShowAdmin           Permissions = "Calendar.Show.Admin"
+	CalendarShowCreator         Permissions = "Calendar.Show.Creator"
+	CalendarSocialAdmin         Permissions = "Calendar.Social.Admin"
+	CalendarSocialCreator       Permissions = "Calendar.Social.Creator"
+	CMSAdmin                    Permissions = "CMS.Admin"
+	CMSEndboardAdmin            Permissions = "CMS.EndboardAdmin"
+	CMSNewsAdmin                Permissions = "CMS.News.Admin"
+	CMSNewsCreator              Permissions = "CMS.News.Creator"
+	CMSNewsItemAdmin            Permissions = "CMS.News.Item.Admin"
+	CMSNewsItemCreator          Permissions = "CMS.News.Item.Creator"
+	CMSPageAdmin                Permissions = "CMS.Page.Admin"
+	CMSPageCreator              Permissions = "CMS.Page.Creator"
+	CMSPermalinkAdmin           Permissions = "CMS.Permalink.Admin"
+	CMSSlideshowAdmin           Permissions = "CMS.Slideshow.Admin"
+	CMSSlideshowCreator         Permissions = "CMS.Slideshow.Creator"
+	CMSView                     Permissions = "CMS.View"
+	Cobra                       Permissions = "COBRA"
+	Director                    Permissions = "Director"
+	EditNetStats                Permissions = "EditNetStats"
+	EmailAccess                 Permissions = "Email.Access"
+	EmailAlumni                 Permissions = "Email.Alumni"
+	EmailEveryone               Permissions = "Email.Everyone"
+	EmailOfficers               Permissions = "Email.Officers"
+	EquipmentAdmin              Permissions = "EquipmentAdmin"
+	HiresAdmin                  Permissions = "HiresAdmin"
+	Inform                      Permissions = "Inform"
+	KeyListManage               Permissions = "KeyList.Manage"
+	MailingListAdmin            Permissions = "MailingList.Admin"
+	ManageMembersAdmin          Permissions = "ManageMembers.Admin"
+	ManageMembersGroup          Permissions = "ManageMembers.Groups"
+	ManageMembersMembersAdd     Permissions = "ManageMembers.Members.Add"
+	ManageMembersMembersAdmin   Permissions = "ManageMembers.Members.Admin"
+	ManageMembersMembersList    Permissions = "ManageMembers.Members.List"
+	ManageMembersMicsKeyList    Permissions = "ManageMembers.Misc.KeyList"
+	ManageMembersMiscUnpaidList Permissions = "ManageMembers.Misc.UnpaidList"
+	ManageMembersOfficers       Permissions = "ManageMembers.Officers"
+	ManageMembersPermissions    Permissions = "ManageMembers.Permissions"
+	MenuDisabled                Permissions = "Menu.Disabled"
+	OfficerReports              Permissions = "OfficerReports"
+	Streamer                    Permissions = "Streamer"
+	SuperUser                   Permissions = "SuperUser"
+	TechieTodo                  Permissions = "TechieTodo"
+	VideoStats                  Permissions = "VideoStats"
+	WatchAdmin                  Permissions = "Watch.Admin"
+)
```

Comment:
Bordering on a nitpick, but what's the purpose of having all these constants - i.e. are they used anywhere in the web-auth codebase, or are they simply here to list all the possible values? If so, this seems a little verbose, and would make acccessing them all more painful (e.g. making a drop-down of all permissions, or validating an input). Recall that if the underlying type of `Permission` is `string`, then I can just do `var p Permission = Permission("THIS_IS_NONSENSE")`, so this doesn't add any type safety.

If the constants aren't actually used, what do you think about turning this into a `[...]Permission` - it'll be less verbose and would also mean that we could get the full list of permissions if we need it somewhere.

Reply:
So I did notice that the constants are indeed used - but I'm not sure that those are the best way of doing things, so I'll let this comment stand for now.

Reply:
I will try and rework it later but for now does work